pallets/registry/src/lib.rs

```diff
@@ -225,18 +259,115 @@ pub mod pallet {
         NoProgramSet,
         TooManyModifiableKeys,
         MismatchedVerifyingKeyLength,
+        JumpStartProgressNotReady,
+        JumpStartNotInProgress,
+        NoRegisteringFromParentKey,
     }
 
+    /// Allows anyone to create a master key for the network if the network is read and a master key
+    /// does not exist
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight({
+            <T as Config>::WeightInfo::jump_start_network()
+        })]
+        pub fn jump_start_network(origin: OriginFor<T>) -> DispatchResult {
+            let _who = ensure_signed(origin)?;
+            let current_block_number = <frame_system::Pallet<T>>::block_number();
+            let converted_block_number: u32 =
+                BlockNumberFor::<T>::try_into(current_block_number).unwrap_or_default();
+
+            // make sure jumpstart is ready, or in progress but X amount of time has passed
+            match JumpStartProgress::<T>::get().jump_start_status {
+                JumpStartStatus::Ready => (),
+                JumpStartStatus::InProgress(started_block_number) => {
+                    if converted_block_number.saturating_sub(started_block_number)
+                        < BLOCKS_TO_RESTART_JUMP_START
+                    {
+                        return Err(Error::<T>::JumpStartProgressNotReady.into());
+                    };
+                },
+                _ => return Err(Error::<T>::JumpStartProgressNotReady.into()),
+            };
+            // TODO: Add checks for network state see https://github.com/entropyxyz/entropy-core/issues/923
+            Dkg::<T>::try_mutate(current_block_number, |messages| -> Result<_, DispatchError> {
+                messages.push(NETWORK_PARENT_KEY.clone().encode());
+                Ok(())
+            })?;
+            JumpStartProgress::<T>::put(JumpStartDetails {
+                jump_start_status: JumpStartStatus::InProgress(converted_block_number),
+                confirmations: vec![],
+            });
+            Self::deposit_event(Event::StartedNetworkJumpStart());
+            Ok(())
+        }
+
+        /// Allows validators to signal a successful network jumpstart
+        #[pallet::call_index(1)]
+        #[pallet::weight({
+                <T as Config>::WeightInfo::jump_start_results_confirm(SIGNING_PARTY_SIZE as u32)
+                .max(<T as Config>::WeightInfo::jump_start_results_done(SIGNING_PARTY_SIZE as u32))
+        })]
+        pub fn jump_start_results(
```

Comment:
This should probably be renamed to something like: 

```suggestion
        pub fn confirm_jump_start(
```